arcanist fails to build on macOS because of failing dependency php-7.1.9

## Issue description
Arcanist fails to build because one of its dependencies, seems to be `php-7.1.9`, fails to build on macOS High Sierra.

I've included the full output [here in this gist](https://gist.github.com/Tehnix/11b40d7169e3a8ddeab4b30b1938da0d). A relevant excerpt,

```
/nix/store/g6l32ncr8vlpprlmv9kgr4qksx929cbj-bash-4.4-p12/bin/bash /private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/libtool --silent --preserve-dup-deps --mode=compile clang++ -I/nix/store/haf5c1ngz99401mzm4123lzhr9jsg538-icu4c-59.1/include  -Wno-write-strings -D__STDC_LIMIT_MACROS -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -Iext/intl/ -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/ext/intl/ -DPHP_ATOM_INC -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/include -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/main -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9 -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/ext/date/lib -I/nix/store/qjkl7l4137b0kb8swdkfai97yjhp2z0z-libxml2-2.9.5-dev/include/libxml2 -I/nix/store/a9al07p241x039155kc059cvp08jahbv-openssl-1.0.2l-dev/include -I/nix/store/61ma58qr8w6h1kcidsr10n2wnzn5ijrk-zlib-1.2.11-dev/include -I/nix/store/k21xbmpchywavnlc3wzva729mx056m11-curl-7.56.0-dev/include -I/nix/store/cgrk26n799wxaz5rx7phzz39y378c7ym-libjpeg-8d-dev/include -I/nix/store/17f6b5w45hm7gwbghricq3avprdazfx5-libpng-apng-1.6.34-dev/include -I/nix/store/g5fxqr2hmcdwldqq1wxnx92siyp4090h-freetype-2.7.1-dev/include/freetype2 -I/nix/store/jyq96j27b3y4iw0zdjnb31av0lg92x67-gettext-0.19.8/include -I/nix/store/7hgnxyff3hnds4zhdyghaa0h5npbr4g2-gmp-6.1.2-dev/include -I/nix/store/l7fsy0fzcndfclnpf0kcklzvv7mjnivw-libiconv-osx-10.11.6/include -I/nix/store/haf5c1ngz99401mzm4123lzhr9jsg538-icu4c-59.1/include -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/ext/mbstring/oniguruma -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/ext/mbstring/libmbfl -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/ext/mbstring/libmbfl/mbfl -I/nix/store/q2nq74f6lw6vgc0w3f7r050djbhnbmw5-libmcrypt-2.5.8/include -I/nix/store/15rncxbk66d71g9gbg99b0z9jj7fkj77-mariadb-client-10.1.28-dev/include/mysql -I/nix/store/15rncxbk66d71g9gbg99b0z9jj7fkj77-mariadb-client-10.1.28-dev/include/mysql/.. -I/nix/store/vqfzhn3bd3rq60004554nnw14s7h23nb-postgresql-9.6.5/include -I/nix/store/hjlclqxi5bcmcrd54fri2a256qxnnzdh-sqlite-3.20.0-dev/include -I/nix/store/wkwp4lbpp6h35p1yv9hpm692ilgk5f7c-readline-6.3p08-dev/include -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/ext/zip/lib -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/TSRM -I/private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/Zend  -no-cpp-precomp  -g -O2  -c /private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/ext/intl/intl_convertcpp.cpp -o ext/intl/intl_convertcpp.lo
In file included from /private/tmp/nix-build-php-7.1.9.drv-0/php-7.1.9/ext/intl/intl_convertcpp.cpp:21:
In file included from ext/intl/intl_convertcpp.h:26:
/nix/store/x88k2myciki60yafz4bxrdsf42mw8wrf-icu4c-59.1-dev/include/unicode/unistr.h:3025:7: error: delegating constructors are permitted only in C++11
      UnicodeString(ConstChar16Ptr(text)) {}
      ^~~~~~~~~~~~~
/nix/store/x88k2myciki60yafz4bxrdsf42mw8wrf-icu4c-59.1-dev/include/unicode/unistr.h:3087:7: error: delegating constructors are permitted only in C++11
      UnicodeString(ConstChar16Ptr(text), length) {}
      ^~~~~~~~~~~~~
/nix/store/x88k2myciki60yafz4bxrdsf42mw8wrf-icu4c-59.1-dev/include/unicode/unistr.h:3180:7: error: delegating constructors are permitted only in C++11
      UnicodeString(Char16Ptr(buffer), buffLength, buffCapacity) {}
      ^~~~~~~~~~~~~
3 errors generated.
make: *** [Makefile:1093: ext/intl/intl_convertcpp.lo] Error 1
builder for ‘/nix/store/g8sig0qlyr2kxifnwg0yf4wb6hm7pwsp-php-7.1.9.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/qyhqpnnr68ir2kaydq58k5glh650kmxs-arcanist-20170323.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/qyhqpnnr68ir2kaydq58k5glh650kmxs-arcanist-20170323.drv’ failed
/nix/var/nix/profiles/default/bin/nix-shell: failed to build all dependencies
```

### Steps to reproduce

Run `nix-shell -p arcanist`

## Technical details

* System: macOS High Sierra 10.13.1 Beta (17B46a)
* Nix version: 1.11.15
* Nixpkgs version: 18.03pre118668.0e4be9e5f0
* Sandboxing enabled: build-use-sandbox = false

--- a/pkgs/development/interpreters/php/default.nix
+++ b/pkgs/development/interpreters/php/default.nix
@@ -20,9 +20,11 @@ let
 
       enableParallelBuilding = true;
 
-  nativeBuildInputs = [ pkgconfig ];
+      nativeBuildInputs = [ pkgconfig ];
       buildInputs = [ flex bison ]
         ++ lib.optional stdenv.isLinux systemd;
+
+      CXXFLAGS = lib.optional stdenv.cc.isClang "-std=c++11";
 
       flags = {
 

